toolchain/check/check_unit.cpp

```diff
@@ -394,6 +396,19 @@ auto CheckUnit::ProcessNodeIds() -> bool {
   return true;
 }
 
+auto CheckUnit::CheckRequiredDeclarations() -> void {
+  for (const auto& function : context_.functions().array_ref()) {
+    if (!function.first_owning_decl_id.has_value() &&
+        function.extern_library_id == context_.sem_ir().library_id()) {
+      // TODO: add condition function.package == context_.sem_ir().package_id()
+      CARBON_DIAGNOSTIC(MissingDeclarationInApi, Error,
```

Comment:
```suggestion
      CARBON_DIAGNOSTIC(MissingOwningDeclarationInApi, Error,
```

I'm hesitant to just call this "missing declaration", how about adding "owning"?